experimental/dds/tree/src/interface-prototype/Examples.ts

```diff
@@ -0,0 +1,141 @@
+import { v4 } from 'uuid';
+// This file uses these as opaque id types:
+// the user of these APIs should not know or care if they are short IDs or not, other than that they must be converted to StableId if stored for use outside of the shared tree it was acquired from.
+// In practice, these would most likely be implemented as ShortId numbers.
+import { Definition, TraitLabel } from '../Identifiers';
+import { Side } from '../Snapshot';
+import { StableId, TreeNodeData } from './Anchors';
+import {
+	anchorDataFromNodeId,
+	Command,
+	CommandContext,
+	CommandId,
+	CommandRegistry,
+	PrefetchFilter,
+	root,
+	SharedTree,
+	TreeNodeViewReadonly,
+} from './Checkout';
+import { Place, TreeNode } from './MutableAnchors';
+
+//////////////// Command examples //////////////
+
+// Inserts a node with the specified Definition and identifier at the specified Place, and return it.
+export const insertExample = {
+	id: 'f73f004c-3b3e-42fe-b7c9-2e5e8793ca45' as CommandId,
+	run: (
+		context: CommandContext,
+		{ definition: def, identifier: id }: { definition: StableId; identifier: StableId },
+		{ place }: { place: Place }
+	): TreeNode => {
+		const definition = context.loadDefinition(def);
+		const identifier = context.loadNodeId(id);
+		const range = place.insert({ definition, identifier, traits: {} });
+		return range.start.adjacentNode(Side.After);
+	},
+};
+
+function newNodeId(): StableId {
+	return v4() as StableId;
+}
+
+export const doubleInsertExample = {
+	id: '08bac27d-632f-48bb-834a-90af8d67ca60' as CommandId,
+	run: (context: CommandContext, _: {}, { parent }: { parent: TreeNode }): TreeNode => {
+		const a = context.runCommand(
+			insertExample,
+			{ definition: context.stabilize(bar), identifier: newNodeId() },
+			{ place: parent.childrenFromTrait(testTrait).start }
+		);
+
+		return context.runCommand(
```

Comment:
I like the API aspects of this, but the specific example might make readers wonder why we didn't just pass a larger "builder" tree in. (I realize this probably isn't the final builder API.)

Reply:
You could pass a large one in, but then either the example would not show nested commands, or would be a larger example.